pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -238,6 +240,7 @@ public void deleteSegmentFile() {
 
   // Segment end criteria
   private volatile long _consumeEndTime = 0;
+  private boolean _endOfPartitionGroup = false;
```

Comment:
Does this need to be `volatile`?

Reply:
Only 1 thread is using this. The same thread that writes it in consumeLoop and then checks it also in consumeLoop. In that case it doesn't need to be volatile rt?

Reply:
consumeLoop is also called in the helix transition thread (while transitioning from CONSUMING to ONLINE)

Reply:
oh right, changed it to volatile then